app/lib/pdf/finance/invoice_generator.rb

```diff
@@ -76,13 +71,17 @@ def print_header
         move_down(3)
       end
 
+      def print_logo
+        @data.with_logo do |logo|
+          @pdf.image(logo, fit: [200,50], position: :left) if logo
+        end
+      end
+
       def print_address_columns
         # TODO: cleanup the constants
-        two_columns( [0.mm, @pdf.cursor], height: 50.mm) do |column|
-          case column
-          when :left then print_address( @data.provider, 'Issued by')
-          when :right then print_address( @data.buyer, 'For')
-          end
+        two_columns([0.mm, @pdf.cursor], height: 50.mm) do |column|
+          print_address(@data.provider, 'Issued by') if column == :left
+          print_address(@data.buyer, 'For') if column == :right
```

Comment:
why refactor this btw?

Reply:
To avoid the warning in the IDE:
```
`case` statement doesn't have an 'else' block
```
Not sure where it comes from though, maybe just a RubyMine thing...
In any case, looks a bit cleaner like this for me.

Reply:
To me it's a little confusing because it is expressed as a series of actions instead of a choice between options. Semantics is lost. The complaint about `else` is because that behavior was not covered, which makes sense as it is a potentially missed handling of input values.

Also I'm not in favor of refactoring expressions just for style as it makes `git blame` harder.

I'm fine if you leave it like this now. In general though we should discuss what style we prefer and makes more sense for us.